Source/Core/DolphinNoGUI/GLFW.h

```diff
@@ -0,0 +1,211 @@
+  }
+  else
+  {
+    rendererHasFocus = false;
+    if (SConfig::GetInstance().bHideCursor)
+      glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_NORMAL);
+  }
+}
+
+static void OnKey(GLFWwindow* window, int key, int scancode, int action, int mods)
+{
+  if (action != GLFW_PRESS)
+    return;
+
+  if (key == GLFW_KEY_ESCAPE)
+  {
+    if (Core::GetState() == Core::CORE_RUN)
+    {
+      if (SConfig::GetInstance().bHideCursor)
+        glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_NORMAL);
+      Core::SetState(Core::CORE_PAUSE);
+    }
+    else
+    {
+      if (SConfig::GetInstance().bHideCursor)
+        glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_HIDDEN);
+      Core::SetState(Core::CORE_RUN);
+    }
+  }
+  else if ((key == GLFW_KEY_ENTER) && (mods & GLFW_MOD_ALT))
+  {
+    GLFWmonitor* monitor = glfwGetWindowMonitor(window);
+    if (monitor)
+    {
+      int last_window_width = SConfig::GetInstance().iRenderWindowWidth;
+      int last_window_height = SConfig::GetInstance().iRenderWindowHeight;
+      glfwSetWindowMonitor(window, nullptr, GLFW_DONT_CARE, GLFW_DONT_CARE, last_window_width,
+                           last_window_height, GLFW_DONT_CARE);
+    }
+    else
+    {
+      monitor = glfwGetPrimaryMonitor();
+      const GLFWvidmode* vidmode = glfwGetVideoMode(monitor);
+      glfwSetWindowMonitor(window, monitor, 0, 0, vidmode->width, vidmode->height,
+                           vidmode->refreshRate);
+    }
+  }
+  else if (key >= GLFW_KEY_F1 && key <= GLFW_KEY_F8)
+  {
+    int slot_number = key - GLFW_KEY_F1 + 1;
+    if (mods & GLFW_MOD_SHIFT)
+      State::Save(slot_number);
+    else
+      State::Load(slot_number);
+  }
+  else if (key == GLFW_KEY_F9)
+    Core::SaveScreenShot();
+  else if (key == GLFW_KEY_F11)
+    State::LoadLastSaved();
+  else if (key == GLFW_KEY_F12)
+  {
+    if (mods & GLFW_MOD_SHIFT)
+      State::UndoLoadState();
+    else
+      State::UndoSaveState();
+  }
+}
+
+static void OnFramebufferSize(GLFWwindow* window, int width, int height)
+{
+  int last_window_width = SConfig::GetInstance().iRenderWindowWidth;
+  int last_window_height = SConfig::GetInstance().iRenderWindowHeight;
+  if (last_window_width != width || last_window_height != height)
+  {
+    last_window_width = width;
+    last_window_height = height;
+
+    // We call Renderer::ChangeSurface here to indicate the size has changed,
+    // but pass the same window handle. This is needed for the Vulkan backend,
+    // otherwise it cannot tell that the window has been resized on some drivers.
+    if (g_renderer)
+      g_renderer->ChangeSurface(s_window_handle);
+  }
+}
+
+class PlatformGLFW : public Platform
+{
+private:
+  GLFWwindow* window = nullptr;
+
+public:
+  void Init() override
+  {
+    glfwSetErrorCallback(OnError);
+
+    if (!glfwInit())
+    {
+      PanicAlert("GLFW initialization failed");
+      exit(1);
+    }
+
+#define LET_S_USE_OPENGL
```

Comment:
This looks a little like testing code, with the `#define` right next to its usage. Should this be towards the top of the file, or in the CMake files instead?